rust/src/doc.rs

```diff
@@ -514,6 +514,13 @@ impl Doc {
             .collect()
     }
 
+    pub fn changes(&self) -> Vec<ChangeHash> {
+        let empty_heads: Vec<am::ChangeHash> = Vec::new();
+        let mut doc = self.0.write().unwrap();
+        let changes = doc.get_changes(&empty_heads);
+        changes.into_iter().map(|h| h.hash().into()).collect()
```

Comment:
The compiler in my head says that this should be `changes.into_iter().cloned().collect()` because the type of `changes` is `Vec<&ChangeHash>` so you need to clone the `&ChangeHash` to get a `ChangeHash`. I've not actually tried to compile it though so I might be completely off base.

Reply:
the `get_changes` returned a Change - not a ChangeHash, which is how I jumped into the above formulation. It does compile in Rust, and then I had clippy and rustfmt yell at me until I tuned the syntax to where it was happy.

Reply:
Oh yes, sleepy brain is sleepy.